File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/android/src/main/kotlin/com/idosmart/model/IDOLiveDataModel.kt
//
//  IDOLiveDataModel.kt
//  protocol_channel
//
//  Created by hedongyang on 2023/10/30.
//

package com.idosmart.model

import com.google.gson.GsonBuilder
import com.google.gson.annotations.SerializedName

///
/// Get Real-time Data event number

open class IDOLiveDataModel(
    totalStep: Int,
    totalCalories: Int,
    totalDistances: Int,
    totalActiveTime: Int,
    heartRate: Int
) : IDOBaseModel {

    /// Total steps
    @SerializedName("total_step")
    var totalStep: Int = totalStep

    /// Total calories (in kilocalories)
    @SerializedName("total_calories")
    var totalCalories: Int = totalCalories

    /// Total distance (in meters)
    @SerializedName("total_distances")
    var totalDistances: Int = totalDistances

    /// Total active time (in seconds)
    @SerializedName("total_active_time")
    var totalActiveTime: Int = totalActiveTime

    /// Heart rate data (in beats per minute)
    @SerializedName("heart_rate")
    var heartRate: Int = heartRate


    

    override fun toJsonString(): String {
        return GsonBuilder().create().toJson(this).toString()
    }
}
    

File: ido_sdk_flutter/ido_sdk/protocol_sdk/protocol_channel/ios/Classes/Model/IDOLiveDataModel.swift
//
//  IDOLiveDataModel.swift
//  protocol_channel
//
//  Created by hc on 2023/10/12.
//

import Foundation
/// Get Real-time Data event number
@objcMembers
public class IDOLiveDataModel: NSObject, IDOBaseModel {
    /// Total steps
    public var totalStep: Int
    /// Total calories (in kilocalories)
    public var totalCalories: Int
    /// Total distance (in meters)
    public var totalDistances: Int
    /// Total active time (in seconds)
    public var totalActiveTime: Int
    /// Heart rate data (in beats per minute)
    public var heartRate: Int
    
    enum CodingKeys: String, CodingKey {
        case totalStep = "total_step"
        case totalCalories = "total_calories"
        case totalDistances = "total_distances"
        case totalActiveTime = "total_active_time"
        case heartRate = "heart_rate"
    }
    
    public init(totalStep: Int,totalCalories: Int,totalDistances: Int,totalActiveTime: Int,heartRate: Int) {
        self.totalStep = totalStep
        self.totalCalories = totalCalories
        self.totalDistances = totalDistances
        self.totalActiveTime = totalActiveTime
        self.heartRate = heartRate
    }
    
    public func toJsonString() -> String? {
        if let jsonData = try? JSONEncoder().encode(self) {
            let jsonString = String(data: jsonData, encoding: .utf8)
            return jsonString
        }
        return nil
    }
}

